test/tests/userobjects/execution_order_groups/tests

```diff
@@ -0,0 +1,36 @@
+[Test]
+  issues = '#23470'
+  design = 'UserObjects/index.md'
+
+  [simultaneous]
+    type = CSVDiff
+    input = test.i
+    cli_args = 'Outputs/file_base=simultaneous'
+    csvdiff = simultaneous.csv
+    requirement = "The system shall execute user objects in the same group in the order given in the design documentation."
+  []
+
+  [two_groups]
+    type = CSVDiff
+    input = test.i
+    cli_args = 'Outputs/file_base=two_groups Postprocessors/pp3/execution_order_group=3'
+    csvdiff = two_groups.csv
+    requirement = "The system shall permit adding one new execution order group."
+  []
+
+  [three_groups]
+    type = CSVDiff
+    input = test.i
+    cli_args = 'Outputs/file_base=three_groups Postprocessors/pp2/execution_order_group=2 Postprocessors/pp3/execution_order_group=3'
+    csvdiff = three_groups.csv
+    requirement = "The system shall permit adding multiple new execution order groups."
+  []
+
+  [negative_group]
+    type = CSVDiff
+    input = test.i
+    cli_args = 'Outputs/file_base=negative_group Postprocessors/pp1/execution_order_group=-1'
```

Comment:
this is the negative test

Reply:
The test is not designed as well as it could. The result depends on the UOs being executed in separate groups, but it does not depend on group _ordering_. I'm working on a fix.